source/RoslynAnalyzers/AnalyzerExtensionMethods.cs

```diff
@@ -47,7 +47,23 @@ public static bool IsAssignableTo(this ITypeSymbol? sourceType, ITypeSymbol? tar
 
             return false;
         }
+        
+        /// <summary>Returns true if sourceType inherits from baseType, walking down the type hierarchy as far as it can</summary>
+        public static bool InheritsFrom(this ITypeSymbol? sourceType, ITypeSymbol? baseType)
+        {
+            if (sourceType?.BaseType is not { } candidate) return false;
+
+            while (candidate != null)
+            {
+                if (SymbolEqualityComparer.Default.Equals(candidate, baseType)) return true;
+
+                candidate = candidate.BaseType?.MetadataName == "Object" ? null : candidate.BaseType;
```

Comment:
This is subjective, but I'd prefer the `candidate.BaseType?.MetadataName != "Object"` in the while condition, and to leave this as a simple assignment.

Reply:
Good suggestion. I've done a refactoring pass based on that suggestion and it's much nicer now, thanks